File: 1137-height-checker/1137-height-checker.java
class Solution {
    public int heightChecker(int[] heights) {
        int[] cnt = new int[101];
        int miss = 0;

        for (int h: heights) cnt[h]++;

        int h = 1, idx = 0;
        while (h < 101) {
            if (cnt[h] == 0) h++;
            else {
                if (h != heights[idx]) miss++;
                idx++;
                cnt[h]--;
            }            
        }

        return miss;
    }
}

File: 1137-height-checker/1137-height-checker.c
int heightChecker(int* heights, int heightsSize) {
    int cnt[101] = {0};
    int* th = heights;
    int miss = 0;

    while (heightsSize--) cnt[*heights++]++;

    int h = 1;
    while (h < 101) {
        if (!cnt[h]) h++;
        else {
            if (h != *th) miss++;
            th++;
            cnt[h]--;
        }
    }

    return miss;
}